common/lib/xmodule/xmodule/modulestore/xml_importer.py

```diff
@@ -1167,5 +1168,5 @@ def _update_module_location(module, new_location):
     # explicitly set each field to its current value before triggering the save.
     if len(rekey_fields) > 0:
         for rekey_field_name in rekey_fields:
-            setattr(module, rekey_field_name, getattr(module, rekey_field_name))
+            module.fields[rekey_field_name].force_dirty(module)
```

Comment:
Blarg! You implemented what I was about to recommend as a fix in https://github.com/edx/XBlock/pull/295, and I talked you into closing it.

I'm sorry I didn't fully understand the issue before now, and have pushed you into such a runaround.

In XBlock-land, it seems like the test that would fail is, with a mutable field, that `block.field = block.field` doesn't trigger a write to the `FieldData` when `block.save()` is called.